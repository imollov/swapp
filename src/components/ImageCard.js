import React from 'react'
import { Flex, Heading, Box, Image } from 'rebass/styled-components'

export default ({ variant, img, title, ...rest }) => {
  const imageWidth = variant === 'compact' ? 1 / 5 : 1 / 3
  return (
    <Flex {...rest} variant="card">
      <Image src={img} width={imageWidth} />
      <Box margin="auto" p={1} textAlign="center">
        <Heading variant="h3">{title}</Heading>
      </Box>
    </Flex>
  )
}
