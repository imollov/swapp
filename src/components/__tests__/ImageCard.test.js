import React from 'react'
import { shallow } from 'enzyme'

import ImageCard from '../ImageCard'

describe('ImageCard Component', () => {
  it('smoke test', () => {
    const wrapper = shallow(
      <ImageCard img="image.png" title="title" linkTo="/" />,
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot with variant', () => {
    const wrapper = shallow(
      <ImageCard variant="compact" img="image.png" title="title" linkTo="/" />,
    )

    expect(wrapper).toMatchSnapshot()
  })
})
