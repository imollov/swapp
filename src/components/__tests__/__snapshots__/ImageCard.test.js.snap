// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard Component should match snapshot with variant 1`] = `
<Styled(styled.div)
  variant="card"
>
  <BackgroundImage
    src="image.png"
    width={0.2}
  />
  <styled.div
    margin="auto"
    p={2}
    textAlign="center"
  >
    <ForwardRef
      variant="h3"
    >
      title
    </ForwardRef>
  </styled.div>
</Styled(styled.div)>
`;

exports[`ImageCard Component smoke test 1`] = `
<Styled(styled.div)
  variant="card"
>
  <BackgroundImage
    src="image.png"
    width={0.3333333333333333}
  />
  <styled.div
    margin="auto"
    p={2}
    textAlign="center"
  >
    <ForwardRef
      variant="h3"
    >
      title
    </ForwardRef>
  </styled.div>
</Styled(styled.div)>
`;
